File: 백준/Bronze/10810. 공 넣기/공 넣기.java
import java.io.*;
import java.util.*;

class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int a = Integer.parseInt(st.nextToken());
        int b = Integer.parseInt(st.nextToken());

        int[] arr = new int[a+1];

        for(int i=0; i<b; i++) {
            st = new StringTokenizer(br.readLine());
            int x = Integer.parseInt(st.nextToken());
            int y = Integer.parseInt(st.nextToken());
            int z = Integer.parseInt(st.nextToken());

            for(int j=x; j<=y; j++) {
                arr[j] = z;
            }

        }

        for(int i=1; i<=a; i++) System.out.print(arr[i] + " ");
    }
}

File: 백준/Bronze/10810. 공 넣기/공 넣기.kt
import java.io.*
import java.util.*

fun main(args: Array<String>) {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val st = StringTokenizer(br.readLine())

    val n = st.nextToken().toInt()
    val m = st.nextToken().toInt()

    var intArray = Array<Int>(n) { 0 }

    for(i in 0 until m) {
        val st = StringTokenizer(br.readLine())

        val a = st.nextToken().toInt()
        val b = st.nextToken().toInt()
        val c = st.nextToken().toInt()

        for(i in a .. b) {
            intArray[i-1] = c
        }
    }

    for(i in intArray) {
        print("$i ")
    }
}